### ISSUE TITLE
the name `GetterSetter` is defined multiple times. Perhaps unhygienic

### ISSUE THREAD

**Issue by Gaareth** [europeanplaice/field_accessor#2]
When using FieldAccessor on multiple structs it results in import errors.

Code:
``` rust
use field_accessor::FieldAccessor;


#[derive(FieldAccessor)]
struct Test {
    pub name: String,
}


#[derive(FieldAccessor)]
struct Test2 {
    pub name: String,
}

```

Error:
```
error[E0428]: the name `GetterSetter` is defined multiple times
 --> src/main.rs:9:10
  |
3 | #[derive(FieldAccessor)]
  |          ------------- previous definition of the trait `GetterSetter` here
...
9 | #[derive(FieldAccessor)]
  |          ^^^^^^^^^^^^^ `GetterSetter` redefined here
  |
  = note: `GetterSetter` must be defined only once in the type namespace of this module
  = note: this error originates in the derive macro `FieldAccessor` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0252]: the name `mem` is defined multiple times
 --> src/main.rs:9:10
  |
3 | #[derive(FieldAccessor)]
  |          ------------- previous import of the module `mem` here
...
9 | #[derive(FieldAccessor)]
  |          ^^^^^^^^^^^^^ `mem` reimported here
  |
  = note: `mem` must be defined only once in the type namespace of this module
  = note: this error originates in the derive macro `FieldAccessor` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0252, E0428.
For more information about an error, try `rustc --explain E0252`.
error: could not compile `macro_debug` due to 2 previous errors

```

I don't know much about macros, but the solutions I can think of are:
- give GetterSetter a name related to the struct. Like `let gettersetter = format_ident!("{}{}", ident, "GetterSetter");`. The `use std::mem;` is still a problem though.
- wrap the struct in a module: 
``` rust 
pub use test::Test;
mod test {
    use field_accessor::FieldAccessor;

    #[derive(FieldAccessor)]
    pub struct Test {
        pub status: String,
    }
} 
```
- split the trait declaration and the specific methods. So create another macro? to expand to the GetterSetter trait and the  `use std::mem;` import. And then use the FieldAccessor derive

I think the user-friendliest option is number 2

**Comment by europeanplaice** [comment]
Great solutions. I fixed these problems by applying your suggestion of `let gettersetter = format_ident!("{}{}", ident, "GetterSetter");`. And for `use std::mem;`, I removed it and replaced `mem` with `std::mem` in the definitions of the functions. I updated this library to `v0.5.2`. Please let me know if you have another issue. Thank you.

**Comment by Gaareth** [comment]
Great, thanks for the quick response
